src/mapWithFeedback.test.ts

```diff
@@ -0,0 +1,209 @@
+import { pipe } from "./pipe";
+import { mapWithFeedback } from "./mapWithFeedback";
+import { createLazyInvocationCounter } from "../test/lazy_invocation_counter";
+import { take } from "./take";
+import { add } from "./add";
+
+describe("data first", () => {
+  describe("base", () => {
+    it("should return an array of successively accumulated values", () => {
+      const result = mapWithFeedback(
+        [1, 2, 3, 4, 5] as const,
+        (acc, x) => acc + x,
+        100,
+      );
+      expect(result).toEqual([101, 103, 106, 110, 115]);
+    });
+
+    it("should use the same accumulator on every iteration if it's mutable, therefore returning an array containing {array length} references to the accumulator.", () => {
+      const results = mapWithFeedback(
+        [1, 2, 3, 4, 5] as const,
+        (acc, x) => {
+          acc[x] = x;
+          return acc;
+        },
+        {} as Record<string, unknown>,
+      );
+
+      const expectedEquality = Array.from({ length: 5 }).fill({
+        "1": 1,
+        "2": 2,
+        "3": 3,
+        "4": 4,
+        "5": 5,
+      });
+      expect(results).toEqual(expectedEquality);
+      const [item] = results;
+      expect(results.every((result) => result === item)).toBe(true);
```

Comment:
This is redundant and over-complex:
```suggestion
      const [item] = results;
      expect(item).toEqual({ "1": 1, "2": 2, "3": 3, "4": 4, "5": 5 });
      for (const result of reslts) {
        expect(result).toBe(item);
      }
```

Reply:
The primary reason for the redundancy is that Vitest throws on the first assertion failure, which could mask some details. I made it into one assertion to have the full error details when the test does fail.

However, I do get your point about the complexity, I wouldn't want to maintain a complex test.